### PR TITLE
[CALCITE-6582] Release Calcite 1.38.0

### PR DESCRIPTION
**Pull request by julianhyde**

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]
This includes fixes from two different issues, (besides the release itself), which is unusual.
It would have been simpler to merge these as separate PR, but I guess this way we don't have to wait for the authors.
There are lots of space-only changes in the history file.

File: site/_docs/history.md [hunk 1]
```diff
@@ -30,39 +30,127 @@ Downloads are available on the
...
+Ruben Quesada Lopez,
+Sergey Nuyanzin,
+Stamatis Zampetakis,
+suibianwanwan,
+Tim Grein,
+TJ Banghart,
+Wegdan Ghazi,
+xiaohei,
+Xiong Duan,
+YiwenWu,
```
Comment (zabetak) [inline site/_docs/history.md:128]: It seems that the name is Yiwen Wu and I found the following entries in the git log:
* YiwenWu <yiwenwu@tencent.com>
* YiwenWu <yiwenwu1006@gmail.com>

File: site/_docs/history.md [hunk 2]
```diff
@@ -78,31 +166,420 @@ override logic to the `getMaxScale` and `getMaxPrecision` methods.
...
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-6427">CALCITE-6427</a>]
+  Use a higher precision for `DECIMAL` intermediate results for some aggregate
+  functions like `STDDEV`
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-5855">CALCITE-5855</a>]
+  Support frame exclusion in window functions, e.g.
+  `OVER (... EXCLUDE CURRENT ROW)`
+* [<a href="https://issues.apache.org/jira/browse/CALCITE-6365">CALCITE-6365</a>]
+  Support `RETURNING` clause of `JSON_QUERY` function
+
+New functions:
```
Comment (zabetak) [inline site/_docs/history.md:203]: nit: This section does not have the usual header highlighting and reference link; not sure if it is intentional.

**Review by zabetak — APPROVED** [review]
Went over the release notes and everything looks good.

**Comment by mihaibudiu** [comment]
@julianhyde CALCITE-6617 has reported two independent problems in a single issue. Here your last commit includes a fix for the second problem. I will create a new JIRA issue and a PR for the first problem reported. Since you considered 6617 a blocker, I will let you decide whether you want to include this fix as well.

File: site/_docs/history.md [hunk 3]
```diff
@@ -30,39 +30,127 @@ Downloads are available on the
...
+Pranava B,
+Rodrigo Rueda,
+Ruben Quesada Lopez,
+Sergey Nuyanzin,
+Stamatis Zampetakis,
+suibianwanwan,
+Tim Grein,
+TJ Banghart,
+Wegdan Ghazi,
+xiaohei,
```
[the remark this answers is not in the document]
Comment (julianhyde) [reply site/_docs/history.md:126]: @suibianwanwank,

Provide a patch to the `.mailmap` file in a reply in this comment thread and I will include it.

Also consider using `git config` to make your identity more consistent. It is helpful for us committers to know which person we are dealing with.

**Comment by julianhyde** [comment]
@mihaibudiu, I just pushed a commit with a couple of test cases - no fix. This is a dev branch and I will squash and re-sequence commits when a fix is available.

The issues in 6617 all seem to be related to comparing fixed-point and floating-point numbers, so I would like a fix for all parts if possible. A good description of the problems will help a lot. Without a good description I am flying blind.

**Comment by mihaibudiu** [comment]
I have filed https://issues.apache.org/jira/browse/CALCITE-6620 and the corresponding fix https://github.com/apache/calcite/pull/4000. 
I will improve the issue https://issues.apache.org/jira/browse/CALCITE-6617 and the corresponding PR https://github.com/apache/calcite/pull/3998 with a better description

File: site/_docs/history.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (suibianwanwank) [reply site/_docs/history.md:126]: Done
`suibianwanwan <suibianwanwan33@foxmail.com> # aka xiaohei`

Thank you, I will do that

[the remark this answers is not in the document]
Comment (julianhyde) [reply site/_docs/history.md:126]: This is what I needed:
```
diff --git a/.mailmap b/.mailmap
index 7ce4e7016..965a3b9fc 100644
--- a/.mailmap
+++ b/.mailmap
@@ -133,6 +133,8 @@ Shuo Cheng <shuo.cs@alibaba-inc.com>
 Slim Bouguerra <slim.bouguerra@gmail.com>
 Stamatis Zampetakis <15013153+zabetak@users.noreply.github.com>
 Stamatis Zampetakis <zabetak@gmail.com>
+suibianwanwan <1597226206@qq.com>
+suibianwanwan <suibianwanwan33@foxmail.com>
 Taras Ledkov <tledkov@gridgain.com>
 Ted Xu (少杰) <shaojie@alibaba-inc.com>
 Venki Korukanti <venki.korukanti@gmail.com>
```

[the remark this answers is not in the document]
Comment (suibianwanwank) [reply site/_docs/history.md]: I got it, Thank for the tweak.

**Pull request closed without merging**